### ISSUE TITLE
VB6 Write with Format statement

### ISSUE THREAD

**Issue by phongkien** [antlr/grammars-v4#3023]
Doesn't look like the grammar support the following
```
Write #fileNum, Format(abc, "###")
```

<img width="374" alt="image" src="https://user-images.githubusercontent.com/450554/213781540-0f8f6ef3-fa24-483e-9e3b-19af2ae42daf.png">

**Comment by phongkien** [comment]
Looks like it is being eaten by DATELITERAL

```
DATELITERAL: HASH (~ [#\r\n])* HASH;
```

to support
```
MyDate = #February 12, 1985#   ' Assign a date.
```

**Comment by bkiers** [comment]
Yes, the rule `DATELITERAL: HASH (~ [#\r\n])* HASH;` is causing `#fileNum, Format(abc, "#` to be tokenized as a `DATELITERAL`.

Looking at the documentation for date literals in VB: https://learn.microsoft.com/en-us/dotnet/visual-basic/language-reference/data-types/date-data-type, it gives the following examples:

- `#5/31/1993#`
- `#1993-5-31#`
- `#1:15:30 PM#`
- `#13:15:30#`
- `#8/13/2002 12:14 PM#`

So the rule `DATELITERAL` might be written like this:

```
DATELITERAL: HASH [0-9A-Z/\-: ]* HASH;
```

in which case the following tokens are created for your input `Write #fileNum, Format(abc, "###")`:

```
WRITE               'Write'
WS                  ' '
FILENUMBER          '#fileNum'
COMMA               ','
WS                  ' '
IDENTIFIER          'Format'
LPAREN              '('
IDENTIFIER          'abc'
COMMA               ','
WS                  ' '
STRINGLITERAL       '"###"'
RPAREN              ')'
EOF                 '<EOF>'
```

But I don't know too much of VB, so I'm not sure if my suggestion is correct.

**Comment by phongkien** [comment]
@bkiers thanks for the suggestions but I think the language support various date format (long, short, ...) and possibly other format as well. Will take your suggestion into consideration.
